t/167-server-rewite.t

```diff
@@ -0,0 +1,259 @@
+    server_rewrite_by_lua_block {
+        ngx.sleep(0.001)
+        ngx.log(ngx.INFO, "server_rewrite_by_lua_block in server")
+    }
+    location /lua {
+        content_by_lua_block {
+            ngx.say("OK")
+        }
+    }
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+server_rewrite_by_lua_block in server
+--- no_error_log
+[error]
+
+
+
+=== TEST 5: ngx.exit(ngx.OK)
+--- config
+    server_rewrite_by_lua_block {
+        ngx.log(ngx.INFO, "ngx.exit")
+        ngx.exit(ngx.OK)
+    }
+    location /lua {
+        content_by_lua_block {
+        ngx.say("OK")
+        }
+    }
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+ngx.exit
+--- no_error_log
+[error]
+
+
+
+=== TEST 6: ngx.exit(503)
+--- config
+    server_rewrite_by_lua_block {
+        ngx.exit(503)
+    }
+    location /lua {
+        content_by_lua_block {
+         ngx.log(ngx.ERR, "content_by_lua")
+         ngx.say("OK")
+        }
+    }
+--- request
+GET /lua
+--- error_code: 503
+--- no_error_log
+[error]
+
+
+
+=== TEST 7: subrequests
+--- config
+    server_rewrite_by_lua_block {
+        ngx.log(ngx.INFO, "is_subrequest:", ngx.is_subrequest)
+    }
+
+    location /lua {
+        content_by_lua_block {
+            local res = ngx.location.capture("/sub")
+            ngx.print(res.body)
+        }
+    }
+
+    location /sub {
+        content_by_lua_block {
+            ngx.say("OK")
+        }
+    }
+
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+is_subrequest:true
+is_subrequest:false
+--- no_error_log
+[error]
+
+
+
+=== TEST 8: rewrite by ngx_http_rewrite_module
+--- config
+    server_rewrite_by_lua_block {
+        ngx.log(ngx.INFO, "uri is ", ngx.var.uri)
+    }
+
+    rewrite ^ /re;
+
+    location /re {
+        content_by_lua_block {
+            ngx.say("RE")
+        }
+    }
+
+    location /ok {
+        content_by_lua_block {
+            ngx.say("OK")
+        }
+    }
+
+--- request
+GET /lua
+--- response_body
+RE
+--- error_log
+uri is /lua
+--- no_error_log
+[error]
+
+
+
+=== TEST 9: exec
+--- config
+    server_rewrite_by_lua_block {
+        if ngx.var.uri ~= "/ok" then
+            ngx.exec("/ok")
+        end
+        ngx.log(ngx.INFO, "uri is ", ngx.var.uri)
+    }
+
+    location /ok {
+        content_by_lua_block {
+            ngx.say("OK")
+        }
+    }
+
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+uri is /ok
+--- no_error_log
+[error]
+
+
+
+=== TEST 10: server_rewrite_by_lua and rewrite_by_lua
+--- http_config
+    server_rewrite_by_lua_block {
+        ngx.log(ngx.INFO, "server_rewrite_by_lua_block in http")
+    }
+--- config
+    location /lua {
+        rewrite_by_lua_block {
+            ngx.log(ngx.INFO, "rewrite_by_lua_block in location")
+        }
+        content_by_lua_block {
+            ngx.say("OK")
+        }
+    }
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+server_rewrite_by_lua_block in http
+rewrite_by_lua_block in location
```

Comment:
need to a test case like this https://github.com/openresty/lua-nginx-module/blob/master/t/041-header-filter.t#L820
 to verify the chunkname.